### ISSUE TITLE
Tests fail with pytest-asyncio 0.17 due to asyncio-mode legacy warning

### ISSUE THREAD

**Issue by bnavigator** [pytest-dev/pytest-mock#272]
pytest-asyncio introduced a new mode parameter and warns if it is set to legacy:

```
[    5s] =================================== FAILURES ===================================
[    5s] ______________________________ test_plain_stopall ______________________________
[    5s] 
[    5s] testdir = <Testdir local('/tmp/pytest-of-abuild/pytest-77/test_plain_stopall0')>
[    5s] 
[    5s]     def test_plain_stopall(testdir: Any) -> None:
[    5s]         """patch.stopall() in a test should not cause an error during unconfigure (#137)"""
[    5s]         testdir.makepyfile(
[    5s]             """
[    5s]             import random
[    5s]     
[    5s]             def get_random_number():
[    5s]                 return random.randint(0, 100)
[    5s]     
[    5s]             def test_get_random_number(mocker):
[    5s]                 patcher = mocker.mock_module.patch("random.randint", lambda x, y: 5)
[    5s]                 patcher.start()
[    5s]                 assert get_random_number() == 5
[    5s]                 mocker.mock_module.patch.stopall()
[    5s]         """
[    5s]         )
[    5s]         result = testdir.runpytest_subprocess()
[    5s] >       result.stdout.fnmatch_lines("* 1 passed in *")
[    5s] E       Failed: nomatch: '* 1 passed in *'
[    5s] E           and: '============================= test session starts =============================='
[    5s] E           and: 'platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-1.0.0'
[    5s] E           and: 'rootdir: /tmp/pytest-of-abuild/pytest-77/test_plain_stopall0'
[    5s] E           and: 'plugins: mock-3.6.1, asyncio-0.17.2'
[    5s] E           and: 'asyncio: mode=legacy'
[    5s] E           and: 'collected 1 item'
[    5s] E           and: ''
[    5s] E           and: 'test_plain_stopall.py .                                                  [100%]'
[    5s] E           and: ''
[    5s] E           and: '=============================== warnings summary ==============================='
[    5s] E           and: '../../../../usr/lib/python3.9/site-packages/pytest_asyncio/plugin.py:191'
[    5s] E           and: "  /usr/lib/python3.9/site-packages/pytest_asyncio/plugin.py:191: DeprecationWarning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file."
[    5s] E           and: '    config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)'
[    5s] E           and: ''
[    5s] E           and: 'test_plain_stopall.py::test_get_random_number'
[    5s] E           and: "  /usr/lib/python3.9/site-packages/pytest_asyncio/plugin.py:317: DeprecationWarning: '@pytest.fixture' is applied to <fixture _mocker, file=/home/abuild/rpmbuild/BUILDROOT/python-pytest-mock-3.6.1-0.x86_64/usr/lib/python3.9/site-packages/pytest_mock/plugin.py, line=388> in 'legacy' mode, please replace it with '@pytest_asyncio.fixture' as a preparation for switching to 'strict' mode (or use 'auto' mode to seamlessly handle all these fixtures as asyncio-driven)."
[    5s] E           and: '    warnings.warn('
[    5s] E           and: ''
[    5s] E           and: '-- Docs: https://docs.pytest.org/en/stable/warnings.html'
[    5s] E           and: '======================== 1 passed, 2 warnings in 0.01s ========================='
[    5s] E       remains unmatched: '* 1 passed in *'
[    5s] 
[    5s] /home/abuild/rpmbuild/BUILD/pytest-mock-3.6.1/tests/test_pytest_mock.py:842: Failed
[    5s] ----------------------------- Captured stdout call -----------------------------
[    5s] running: /usr/bin/python3.9 -mpytest --basetemp=/tmp/pytest-of-abuild/pytest-77/test_plain_stopall0/runpytest-0
[    5s]      in: /tmp/pytest-of-abuild/pytest-77/test_plain_stopall0
[    5s] ============================= test session starts ==============================
[    5s] platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
[    5s] rootdir: /tmp/pytest-of-abuild/pytest-77/test_plain_stopall0
[    5s] plugins: mock-3.6.1, asyncio-0.17.2
[    5s] asyncio: mode=legacy
[    5s] collected 1 item
[    5s] 
[    5s] test_plain_stopall.py .                                                  [100%]
[    5s] 
[    5s] =============================== warnings summary ===============================
[    5s] ../../../../usr/lib/python3.9/site-packages/pytest_asyncio/plugin.py:191
[    5s]   /usr/lib/python3.9/site-packages/pytest_asyncio/plugin.py:191: DeprecationWarning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
[    5s]     config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)
[    5s] 
[    5s] test_plain_stopall.py::test_get_random_number
[    5s]   /usr/lib/python3.9/site-packages/pytest_asyncio/plugin.py:317: DeprecationWarning: '@pytest.fixture' is applied to <fixture _mocker, file=/home/abuild/rpmbuild/BUILDROOT/python-pytest-mock-3.6.1-0.x86_64/usr/lib/python3.9/site-packages/pytest_mock/plugin.py, line=388> in 'legacy' mode, please replace it with '@pytest_asyncio.fixture' as a preparation for switching to 'strict' mode (or use 'auto' mode to seamlessly handle all these fixtures as asyncio-driven).
[    5s]     warnings.warn(
[    5s] 
[    5s] -- Docs: https://docs.pytest.org/en/stable/warnings.html
[    5s] ======================== 1 passed, 2 warnings in 0.01s =========================

...

[    5s] =========================== short test summary info ============================
[    5s] SKIPPED [4] tests/test_pytest_mock.py:38: this test needs assertion rewrite to work but current option is "plain"
[    5s] FAILED tests/test_pytest_mock.py::test_plain_stopall - Failed: nomatch: '* 1 ...
[    5s] FAILED tests/test_pytest_mock.py::test_used_with_class_scope - Failed: nomatc...
[    5s] FAILED tests/test_pytest_mock.py::test_used_with_module_scope - Failed: nomat...
[    5s] FAILED tests/test_pytest_mock.py::test_used_with_package_scope - Failed: noma...
[    5s] FAILED tests/test_pytest_mock.py::test_used_with_session_scope - Failed: noma...
[    5s] =================== 5 failed, 63 passed, 4 skipped in 2.07s ====================
```

These tests expect a testdir.pytest_subprocess run without warnings.

The easiest way is to patch the matching assertion
```diff
-    result.stdout.fnmatch_lines("* 1 passed in *")
+    result.stdout.fnmatch_lines("* 1 passed")
```
but I think it would be better to provide an explicit asyncio mode configuration to the testdir/pytester fixture.

**Comment by nicoddemus** [comment]
Thanks @bnavigator! Fixed in the latest `master`.

**Comment by Tom-python0121** [comment]
@nicoddemus When will a new version be released?

**Comment by nicoddemus** [comment]
Just released `3.7.0`. 👍
